tests/tests_default.yml

```diff
@@ -0,0 +1,11 @@
+
+- name: Ensure that the role runs with default parameters
+  hosts: all
+  become: true
+
+  tasks:
+    - name: default run (NOOP)
+      vars:
+        logging_role_path: "roles/linux-system-roles.logging"
```

Comment:
What is this parameter and why is it needed? Does every user of the role need to set such a parameter?

Reply:
@pcahyna oVirt needs this parameter. 
Since the role is not yet available in RHEL, We copy it from the repository during build time to the oVirt metrics role path as a sub role.
We use this to define the custom path.

Reply:
@pcahyna, @sradco 
>> What is this parameter and why is it needed? Does every user of the role need to set such a parameter?

> Since the role is not yet available in RHEL, We copy it from the repository during build time to the oVirt metrics role path as a sub role.

For the test scripts, unless we set the parameter to give the path from the top, the tests fail like this:
```
TASK [include Rsyslog role] **************************************************************************************
task path: /home/nhosoi/linux-system-roles/logging/tests/roles/linux-system-roles.logging/tasks/main.yaml:132
ERROR! the role 'logging/roles/rsyslog' was not found in /path/to/linux-system-roles/logging/tests/roles:/path/to/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/path/to/linux-system-roles/logging/tests
```
The failed location is here:
```
132     - name: include Rsyslog role
133       include_role:
134         name: "{{ logging_role_path|d('logging') }}/roles/rsyslog"
```
Could there be any better way to solve/workaround the ERROR?

Reply:
@nhosoi You can set "roles/linux-system-roles.logging" as the default value for this variable

Reply:
@sradco,
> @nhosoi You can set "roles/linux-system-roles.logging" as the default value for this variable

I also thought we could, but if "roles/linux-system-roles.logging" is set as default, the standalone ansible-playbook fails like this [1].  Of course, when I run the stansalone ansible-playbook, if I set "logging_role_path: logging" in my vars, ansible deploys the config files just fine.  That is, I'd expect you need to add "logging_role_path: logging" to the ovirt var file, as well.  Can I have your thoughts who should make this adjustment?  IMHO, if the Logging role is once available in RHEL, the problem disappears; I'd like the CI tests to have "roles/linux-system-roles.logging".
```
+++ b/defaults/main.yml
.....................
+logging_role_path: "roles/linux-system-roles.logging"
```
[1]
```
TASK [include Rsyslog role] **************************************************************************************
task path: /path/to/linux-system-roles/logging/tasks/main.yaml:132
ERROR! the role 'roles/linux-system-roles.logging/roles/rsyslog' was not found in /path/to/linux-system-roles/roles:/path/to/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/path/to/linux-system-roles
```

Reply:
@sradco to me it seems not nice to have his parameter, roles should know this themselves if needed, users should not have to tel them where they are. Can't you use `role_path` or `role_name` or `ansible_parent_role_names` or `ansible_parent_role_paths` https://docs.ansible.com/ansible/latest/reference_appendices/special_variables.html#magic ?

Reply:
@pcahyna, thanks for your inputs!  Let me try them to see if we could avoid using logging_role_path.